### PR TITLE
SDSS-1314: Drupal 10.3 / PHP 8.3

### PR DESCRIPTION
**Pull request by joegl**
# READY FOR REVIEW

# Summary
- Upgrade to Drupal 10.3 and PHP 8.3

# Review Tasks

## Setup tasks and/or behavior to test

1. Focus on code review and not so much local review
2. The upgrade has been deployed to the dev environment and sites can be reviewed there.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[SDSS-1314](https://stanfordits.atlassian.net/browse/SDSS-1314): DEV | Upgrade to Drupal 10.3.x and PHP 8.3

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

[SDSS-1314]: https://stanfordits.atlassian.net/browse/SDSS-1314?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docroot/profiles/sdss/sdss_profile/modules/sdss_entities/src/Form/SdssEntityTypeForm.php [hunk 1]
```diff
@@ -73,7 +73,9 @@ public function save(array $form, FormStateInterface $form_state) {
     elseif ($status == SAVED_NEW) {
       $message = $this->t('The SDSS entity type %name has been added.', $t_args);
     }
-    $this->messenger()->addStatus($message);
+    if ($message) {
```
Comment (mdyoung3) [inline docroot/profiles/sdss/sdss_profile/modules/sdss_entities/src/Form/SdssEntityTypeForm.php:76]: @joegl You might want to initialize this variable.

File: docroot/profiles/sdss/sdss_profile/sdss_profile.post_update.php [hunk 2]
```diff
@@ -5,106 +5,3 @@
...
  */
 
-use Drupal\block_content\Entity\BlockContent;
-use Drupal\block\Entity\Block;
-use Drupal\Core\Serialization\Yaml;
-
-/**
- * Disable the core search module.
- */
-function sdss_profile_post_update_8200() {
```
Comment (pookmish) [inline docroot/profiles/sdss/sdss_profile/sdss_profile.post_update.php:15]: Make sure to add in a `hook_removed_post_updates` when you remove these.

File: patches/contrib/node_revision_delete-mr-26.patch [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
++# Docs at https://git.drupalcode.org/project/gitlab_templates/-/blob/main/includes/include.drupalci.variables.yml
++################
++# variables:
++#   SKIP_ESLINT: '1'
++#   OPT_IN_TEST_NEXT_MAJOR: '1'
++#   _CURL_TEMPLATES_REF: 'main'
++
++###################################################################################
++#
++#                                        *
```
Comment (pookmish) [inline patches/contrib/node_revision_delete-mr-26.patch:70]: Wat? a big drupal drop in a patch. :shrug:

**Review by pookmish — COMMENTED** [review]
I only saw the 1 issue with a high level review.

File: patches/contrib/node_revision_delete-mr-26.patch (hunk 3 above)
Comment (joegl) [reply patches/contrib/node_revision_delete-mr-26.patch:70]: Yea the merge request also adds in the gitlab-ci.yml 🤷

File: docroot/profiles/sdss/sdss_profile/sdss_profile.post_update.php (hunk 2 above)
Comment (joegl) [reply docroot/profiles/sdss/sdss_profile/sdss_profile.post_update.php:15]: Added 👍

**Pull request merged**